File: HomeAutomation/HomeAutomation/src/main/Code/components/VoiceCommandProcessor.java
package components;

import core.EventType;
import core.HomeSystem;
import core.HomeSystemObserver;
import java.util.List;
import java.util.Map;

public interface VoiceCommandProcessor extends HomeSystemObserver {
    void processCommand(String command);
    List<String> getCommandHistory();
    Map<String, Object> parseCommand(String command);
}

File: HomeAutomation/HomeAutomation/Components/VoiceCommandProcessor.cs
using HomeAutomation.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HomeAutomation.Core;


namespace HomeAutomation.Components
{
    public class VoiceCommandProcessor : IComponent
    {
        public List<string> CommandHistory { get; private set; } = new List<string>();
        private VoiceCommandHandler _handlerChain;

        public VoiceCommandProcessor()
        {
          
            var lightHandler = new LightCommandHandler();
            var temperatureHandler = new TemperatureCommandHandler();
            var defaultHandler = new DefaultCommandHandler();

            lightHandler.SetNext(temperatureHandler);
            temperatureHandler.SetNext(defaultHandler);

            _handlerChain = lightHandler;
        }

        public string ProcessCommand(string command)
        {
            CommandHistory.Add(command);
            return _handlerChain.HandleCommand(command);
        }

       
        public void OnStateChanged(EnvironmentalState state)
        {
            
            
        }
    }
}
